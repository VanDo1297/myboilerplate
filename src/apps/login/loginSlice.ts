import { createSlice } from '@reduxjs/toolkit';
import { RootState } from '../../reducer';

// Types
export type CounterState = {
    email: string,
    error:string,
    loading: boolean
}

const initialState1: CounterState = {
    email: '',
    error:'',
    loading: false
}

export const authSlice = createSlice({
    name: 'auth',
    initialState: initialState1,
    reducers: {
        login: (state, action) => {
            state.email = action.payload.email
        },
    },
})

// Slice action creators
export const { login } = authSlice.actions

export const selectAuth = (state: RootState) => state.auth
// export const selectAuth = (state: RootState) => state.auth

export const authReducer =  authSlice.reducer