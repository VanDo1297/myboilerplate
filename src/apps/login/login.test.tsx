import React from 'react';
import LoginPage from './login';
import { render , fireEvent} from '@testing-library/react';
import {CounterState, selectAuth} from './loginSlice';
import { Provider, useSelector, useDispatch } from 'react-redux'
import configureStore from 'redux-mock-store';
import * as reactRedux from 'react-redux';

describe('<LoginPage />', () => {
    const useSelectorMock = jest.spyOn(reactRedux, 'useSelector')

    const initialState = {
        email:'docs2gtvt@gmail.com'
    } as CounterState;
    const mockStore = configureStore();
   
    useSelectorMock.mockReturnValue({
        email:''
    })

    it('test render with input value is dotesing',()=>{
        const store = mockStore(initialState)
        const wrapper = render(<Provider store={store}><LoginPage /></Provider>)
        const {baseElement} = wrapper
        const inputs = baseElement.querySelector('input') as Element;
        fireEvent.change(inputs, {
            target: { value: 'dotesing' },
        });
        expect(wrapper).toMatchSnapshot();
    })

    it('test render with input value is dotesing@gmail.con and default value selector is empty',()=>{
        useSelectorMock.mockClear()
        const store = mockStore(initialState)
        const wrapper = render(<Provider store={store}><LoginPage /></Provider>)
        const {baseElement, getByText} = wrapper
        const dispatchMock = jest.fn();
        dispatchMock.mockImplementation(action => store.dispatch(action));
        
        const inputs = baseElement.querySelector('input') as Element;
        fireEvent.change(inputs, {
            target: { value: 'dotesing@gmail.com' },
        });
        fireEvent.click(getByText('Sumbit'));
        expect(wrapper).toMatchSnapshot();
    })

    it('should render button', () => {
        const store = mockStore(initialState)
        const wrapper = render(<Provider store={store}><LoginPage /></Provider>)
        expect(wrapper).toMatchSnapshot();
    });

    it('test render after click reset',()=>{
        const store = mockStore(initialState)
        const wrapper = render(<Provider store={store}><LoginPage /></Provider>)
        const {baseElement, getByText} = wrapper
        const dispatchMock = jest.fn();
        dispatchMock.mockImplementation(action => store.dispatch(action));
        
        const inputs = baseElement.querySelector('input') as Element;
        fireEvent.change(inputs, {
            target: { value: 'dotesing1212@gmail.com' },
        });
        fireEvent.click(getByText('Reset'));
        expect(wrapper).toMatchSnapshot();
    })
});

  
