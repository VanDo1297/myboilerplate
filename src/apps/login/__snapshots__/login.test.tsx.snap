// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage /> redering1 test render after click reset 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="login__container"
      >
        <p
          class="login__title"
        >
          Login page
        </p>
        <div
          class="login__content"
        >
          <input
            id=""
            name="email"
            placeholder="Enter email..."
            type="text"
            value=""
          />
          <button>
            Sumbit
          </button>
        </div>
        <div
          class="login__result"
        >
          <p
            class=""
          >
            This is email : 
          </p>
          <p>
            asdasd
          </p>
        </div>
        <div
          class="login__result login__content"
        >
          <p
            class=""
          >
            This is old email : 
            dotesing1212@gmail.com
          </p>
          <button>
            Reset
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="login__container"
    >
      <p
        class="login__title"
      >
        Login page
      </p>
      <div
        class="login__content"
      >
        <input
          id=""
          name="email"
          placeholder="Enter email..."
          type="text"
          value=""
        />
        <button>
          Sumbit
        </button>
      </div>
      <div
        class="login__result"
      >
        <p
          class=""
        >
          This is email : 
        </p>
        <p>
          asdasd
        </p>
      </div>
      <div
        class="login__result login__content"
      >
        <p
          class=""
        >
          This is old email : 
          dotesing1212@gmail.com
        </p>
        <button>
          Reset
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<LoginPage /> rendering should render button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="login__container"
      >
        <p
          class="login__title"
        >
          Login page
        </p>
        <div
          class="login__content"
        >
          <input
            id=""
            name="email"
            placeholder="Enter email..."
            type="text"
            value=""
          />
          <button>
            Sumbit
          </button>
        </div>
        <div
          class="login__result"
        >
          <p
            class=""
          >
            This is email : 
          </p>
          <p>
            asdasd
          </p>
        </div>
        <div
          class="login__result login__content"
        >
          <p
            class=""
          >
            This is old email : 
            
          </p>
          <button>
            Reset
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="login__container"
    >
      <p
        class="login__title"
      >
        Login page
      </p>
      <div
        class="login__content"
      >
        <input
          id=""
          name="email"
          placeholder="Enter email..."
          type="text"
          value=""
        />
        <button>
          Sumbit
        </button>
      </div>
      <div
        class="login__result"
      >
        <p
          class=""
        >
          This is email : 
        </p>
        <p>
          asdasd
        </p>
      </div>
      <div
        class="login__result login__content"
      >
        <p
          class=""
        >
          This is old email : 
          
        </p>
        <button>
          Reset
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<LoginPage /> should render button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="login__container"
      >
        <p
          class="login__title"
        >
          Login page
        </p>
        <div
          class="login__content"
        >
          <input
            id=""
            name="email"
            placeholder="Enter email..."
            type="text"
            value=""
          />
          <button>
            Sumbit
          </button>
        </div>
        <div
          class="login__result"
        >
          <p
            class=""
          >
            This is email : 
          </p>
          <p>
            asdasd
          </p>
        </div>
        <div
          class="login__result login__content"
        >
          <p
            class=""
          >
            This is old email : 
            
          </p>
          <button>
            Reset
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="login__container"
    >
      <p
        class="login__title"
      >
        Login page
      </p>
      <div
        class="login__content"
      >
        <input
          id=""
          name="email"
          placeholder="Enter email..."
          type="text"
          value=""
        />
        <button>
          Sumbit
        </button>
      </div>
      <div
        class="login__result"
      >
        <p
          class=""
        >
          This is email : 
        </p>
        <p>
          asdasd
        </p>
      </div>
      <div
        class="login__result login__content"
      >
        <p
          class=""
        >
          This is old email : 
          
        </p>
        <button>
          Reset
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<LoginPage /> test render after click reset 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="login__container"
      >
        <p
          class="login__title"
        >
          Login page
        </p>
        <div
          class="login__content"
        >
          <input
            id=""
            name="email"
            placeholder="Enter email..."
            type="text"
            value=""
          />
          <button>
            Sumbit
          </button>
        </div>
        <div
          class="login__result"
        >
          <p
            class=""
          >
            This is email : 
          </p>
          <p>
            asdasd
          </p>
        </div>
        <div
          class="login__result login__content"
        >
          <p
            class=""
          >
            This is old email : 
            dotesing1212@gmail.com
          </p>
          <button>
            Reset
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="login__container"
    >
      <p
        class="login__title"
      >
        Login page
      </p>
      <div
        class="login__content"
      >
        <input
          id=""
          name="email"
          placeholder="Enter email..."
          type="text"
          value=""
        />
        <button>
          Sumbit
        </button>
      </div>
      <div
        class="login__result"
      >
        <p
          class=""
        >
          This is email : 
        </p>
        <p>
          asdasd
        </p>
      </div>
      <div
        class="login__result login__content"
      >
        <p
          class=""
        >
          This is old email : 
          dotesing1212@gmail.com
        </p>
        <button>
          Reset
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<LoginPage /> test render with input value is dotesing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="login__container"
      >
        <p
          class="login__title"
        >
          Login page
        </p>
        <div
          class="login__content"
        >
          <input
            id=""
            name="email"
            placeholder="Enter email..."
            type="text"
            value="dotesing"
          />
          <button>
            Sumbit
          </button>
        </div>
        <div
          class="login__result"
        >
          <p
            class=""
          >
            This is email : 
          </p>
          <p />
        </div>
        <div
          class="login__result login__content"
        >
          <p
            class=""
          >
            This is old email : 
            
          </p>
          <button>
            Reset
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="login__container"
    >
      <p
        class="login__title"
      >
        Login page
      </p>
      <div
        class="login__content"
      >
        <input
          id=""
          name="email"
          placeholder="Enter email..."
          type="text"
          value="dotesing"
        />
        <button>
          Sumbit
        </button>
      </div>
      <div
        class="login__result"
      >
        <p
          class=""
        >
          This is email : 
        </p>
        <p />
      </div>
      <div
        class="login__result login__content"
      >
        <p
          class=""
        >
          This is old email : 
          
        </p>
        <button>
          Reset
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<LoginPage /> test render with input value is dotesing@gmail.con and default value selector is empty 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="login__container"
      >
        <p
          class="login__title"
        >
          Login page
        </p>
        <div
          class="login__content"
        >
          <input
            id=""
            name="email"
            placeholder="Enter email..."
            type="text"
            value="dotesing@gmail.com"
          />
          <button>
            Sumbit
          </button>
        </div>
        <div
          class="login__result"
        >
          <p
            class=""
          >
            This is email : 
          </p>
          <p />
        </div>
        <div
          class="login__result login__content"
        >
          <p
            class=""
          >
            This is old email : 
            
          </p>
          <button>
            Reset
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="login__container"
    >
      <p
        class="login__title"
      >
        Login page
      </p>
      <div
        class="login__content"
      >
        <input
          id=""
          name="email"
          placeholder="Enter email..."
          type="text"
          value="dotesing@gmail.com"
        />
        <button>
          Sumbit
        </button>
      </div>
      <div
        class="login__result"
      >
        <p
          class=""
        >
          This is email : 
        </p>
        <p />
      </div>
      <div
        class="login__result login__content"
      >
        <p
          class=""
        >
          This is old email : 
          
        </p>
        <button>
          Reset
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
